packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -334,11 +336,67 @@ abstract class IntelliJValidator extends DoctorValidator {
       return;
     }
 
+    
+
     messages.add(new ValidationMessage.error(
       '$title plugin not installed; this adds $title specific functionality.'
     ));
   }
 
+  Future<Null> _validateFlutterPackage(List<ValidationMessage> messages, List<String> packageNames, String title, {
+    Version minVersion
+  }) async {
+
+    for (String packageName in packageNames) {
+      if (!hasPackage(packageName)) {
+        continue;
+      }
+      final String versionText = _readPackageVersion(packageName);
+      final Version localVersion = new Version.parse(versionText);
+      if (localVersion != null && minVersion != null && localVersion < minVersion) {
+        messages.add(new ValidationMessage.error(
+          '$title plugin version $versionText - the recommended minimum version is $minVersion'
+        ));
+      } 
+      else {
+        
+        try {
+          final String latestFlutter = await _getLatestVersionPossible();
+          final Version flutterVersion = new Version.parse(latestFlutter);
+          final String serverMessage = localVersion < flutterVersion ? "[Consider upgrading your plugin version to $version]" : "";
+            messages.add(new ValidationMessage(
+              '$title plugin ${localVersion != null ? "version $localVersion" : "installed"} $serverMessage'
+            ));
+        } catch (e) {
+          messages.add(new ValidationMessage(
+            '$title plugin ${localVersion != null ? "version $localVersion" : "installed"}'
+          ));
+        }
+      }
+
+      return;
+    }
+    messages.add(new ValidationMessage.error(
+      '$title plugin not installed; this adds $title specific functionality.'
+    ));
+  }
+
+  Future<String> _getLatestVersionPossible() async {
+      
+      final List<String> stripVersion = version.split(".");
+      final String buildNumber = '${stripVersion[0].substring(2)}${stripVersion[1]}.${stripVersion[2]}';
+      final String url = 'https://plugins.jetbrains.com/plugins/list?build=IU-$buildNumber&pluginId=io.flutter';
+      
+      try {
+        final http.Response response = await http.get(url);
+        final xml.XmlDocument xmlDocument = xml.parse(response.body);
+        final String version = xmlDocument.findAllElements('version').map((xml.XmlElement node) => node.text).first;
+        return version;
+      } catch (e) {
+        return e;
```

Comment:
At a quick glance, it seems like maybe we want to handle exceptions here as you do above (with a `ValidationMessage`?)

Reply:
Not sure if you want a ValidationMessage for an error returning from the server. If the user is offline then it will always give an error. Currently im just ignoring if the server dosent respond.  Would you like an error message anyway? to say that it couldnt ready the intellij server?

Reply:
Good point.  Something indicating that the server couldn't be reached would probably be nice.

@mit-mit ?